### DISCUSSION TITLE
TypeError: register_adapter() takes 2 positional arguments but 3 were given

### DISCUSSION THREAD

**Discussion by tusugit** [ZelKnow/Hearthbot#9] [Q&A] [answered]
![屏幕截图 2022-04-15 112043](https://user-images.githubusercontent.com/94559152/163513032-0e4a6e54-42c7-473b-87e4-b09ee0becc3d.jpg)
又卡住了，看上去是在注册时应该传两个参，但是却传了三个参。找了半天也没找到问题所在，求解。
有没有可能是我安装go -cqhttp时选择12的原因？

**Comment by ZelKnow** [comment, marked as answer]
看起来是 nonebot2 的版本问题，看一下你的环境下 nonebot2 的版本是多少

**Reply by tusugit** [reply]
版本似乎是1.8，好像不是nonebot2只是nonebot吗？

**Reply by ZelKnow** [reply]
nonebot2 应该没有 1.8 版本，你去 `poetry.lock` 里搜索一下，例如我这里是
```
[[package]]
name = "nonebot2"
version = "2.0.0a16"
description = "An asynchronous python bot framework."
category = "main"
optional = false
python-versions = ">=3.7.3,<4.0.0"
```
代表我的 nonebot2 版本是 2.0.0a16

**Reply by tusugit** [reply]
我下载的是您的源码，与您发的一致。我刚刚说nonebot是1.8版本，是因为我在控制台输入pip uninstall nonebot时，显示查找到1.8版本的nonebot是否卸载。

**Reply by ZelKnow** [reply]
是不是你不是用 `poetry` 虚拟环境去跑的？`nonebot2` 最新的 beta 版本改了 API，`register_adapter` 函数改成只接收参数 `adapter` 了，之前还会接收一个 `name`（[参考这里](https://github.com/nonebot/nonebot2/blob/2f3324ce0c27c9c98a41709901f87c6721a27f39/nonebot/internal/driver/driver.py#L50)），所以你跑的时候使用的版本可能是新的 beta 版

**Reply by ZelKnow** [reply]
使用命令 `poetry run nb run` 去运行机器人，如果你的 `poetry.lock` 与源码一致，应该不会出现这个问题

**Reply by tusugit** [reply]
久等了，感谢指点，我确实是用poetry跑的，我把项目删除之后重新git了一次，python也重新安装了最新版本，目前可以正常运行了。我猜测可能是依赖出现了故障，因为我刚开始的python是3.7版本的，而此项目最低也需要3.8，我卸载3.7重新安装3.8之后出现了上述错误，可能存在依赖没有删除干净的原因。我将python再次卸载，文件也尽可能删除之后下载python3.10版本之后可以正常使用。

另外我是第一次接触这个，想请教一下go -cqhttp的详细设置，例如下图中安装时的5个通信方式应该怎么选，打扰了
![30cea2b6baa147f49a77fd91af7cf5a6](https://user-images.githubusercontent.com/94559152/163561616-3e2546be-66b1-4e8a-aa39-d51567c52cc3.png)

**Reply by ZelKnow** [reply]
我一般是使用 3
